flat_mutation_reader.hh

```diff
@@ -175,6 +176,8 @@ public:
         bool is_end_of_stream() const { return _end_of_stream; }
         bool is_buffer_empty() const { return _buffer.empty(); }
         bool is_buffer_full() const { return _buffer_size >= max_buffer_size_in_bytes; }
+        bool is_close_required() const { return _close_required; }
+        void set_close_required() { _close_required = true; }
```

Comment:
Maybe we can have flat_mutation_reader::fill_buffer() (and similar) set the flag unconditionally (and keep the flag in flat_mutation_reader::impl).

Reply:
Exactly that appears to be the case already, unless I misunderstand

Reply:
Yes, somehow I thought individual readers touch the flag.